app/controllers/campaign_pages_controller.rb

```diff
@@ -43,6 +55,42 @@ def create
   end
 
   def show
-    @page = CampaignPage.find params[:id]
+    @campaign_page = CampaignPage.find params[:id]
+    if @campaign_page.active == false
+      redirect_to :campaign_pages, notice: "The page you wanted to view has been deactivated."
+    end  
   end
+
+  def edit
+    @campaign_page = CampaignPage.find params[:id]
+  end
+
+  def update
```

Comment:
It seems like a lot of the code for update and create is now shared (or close to shared). Is this something we could refactor into being a single method to clean out the widgets before dropping them into the attributes array?